File: test_websocket.py
#!/usr/bin/env python3
import websockets
import asyncio
import json

async def test_session():
    try:
        async with websockets.connect('ws://localhost:6969/ws/session') as ws:
            print("Connected to WebSocket")
            
            # Wait for handshake message
            message = await ws.recv()
            print(f"Received: {message}")
            
            # Send a test message
            test_msg = json.dumps({"type": "input", "data": "help\n"})
            await ws.send(test_msg)
            print("Sent help command")
            
            # Wait for response
            response = await ws.recv()
            print(f"Response: {response}")
            
    except Exception as e:
        print(f"Error: {e}")

if __name__ == "__main__":
    asyncio.run(test_session())

File: test_websocket.js
const WebSocket = require('ws');

// Test WebSocket connections
async function testWebSocket(url, name) {
  console.log(`Testing ${name} WebSocket: ${url}`);
  
  return new Promise((resolve, reject) => {
    const ws = new WebSocket(url);
    
    ws.on('open', () => {
      console.log(`✓ ${name} WebSocket connected successfully`);
      ws.close();
      resolve(true);
    });
    
    ws.on('message', (data) => {
      console.log(`📨 ${name} received:`, data.toString());
    });
    
    ws.on('error', (error) => {
      console.log(`✗ ${name} WebSocket error:`, error.message);
      reject(error);
    });
    
    ws.on('close', (code, reason) => {
      console.log(`🔌 ${name} WebSocket closed:`, code, reason.toString());
    });
    
    // Timeout after 5 seconds
    setTimeout(() => {
      if (ws.readyState !== WebSocket.OPEN) {
        console.log(`⏰ ${name} WebSocket connection timeout`);
        ws.close();
        reject(new Error('Connection timeout'));
      }
    }, 5000);
  });
}

async function testAll() {
  try {
    console.log('Testing WebSocket connections to localhost:6969...\n');
    
    await testWebSocket('ws://localhost:6969/ws/logs/combined', 'Logs Combined');
    console.log('');
    
    await testWebSocket('ws://localhost:6969/ws/session', 'Session');
    console.log('');
    
    console.log('✅ All WebSocket tests passed!');
    process.exit(0);
  } catch (error) {
    console.log('❌ WebSocket test failed:', error.message);
    process.exit(1);
  }
}

testAll();